retrofit/src/main/java/retrofit2/Retrofit.java

```diff
@@ -64,16 +64,19 @@
   final HttpUrl baseUrl;
   final List<Converter.Factory> converterFactories;
   final List<CallAdapter.Factory> callAdapterFactories;
+  final int userAddedCallAdapterFactoryCount;
```

Comment:
ugh, i can't think of a better way to do this.

Reply:
maybe have the platform tell you how many callAdapterFactories it produces? Then you keep this state once in Platform rather than in each instance of `Retrofit`

Reply:
yeah 👍